File: app/src/main/java/com/ng/ngleetcode/view/adapter/NodeTreeAdapter.java
package com.ng.ngleetcode.view.adapter;

import com.chad.library.adapter.base.BaseNodeAdapter;
import com.chad.library.adapter.base.entity.node.BaseNode;
import com.ng.code.util.CodeBean;
import com.ng.code.util.tree.CodeDirNode;
import com.ng.code.util.tree.CodeNode;
import com.ng.code.util.tree.HeadNode;
import com.ng.ngleetcode.view.provider.CodeDirProvider;
import com.ng.ngleetcode.view.provider.CodeProvider;
import com.ng.ngleetcode.view.provider.HeadProvider;

import org.jetbrains.annotations.NotNull;

import java.util.List;

public class NodeTreeAdapter extends BaseNodeAdapter {

    private BaseNode nowItem;

    public NodeTreeAdapter(OnLeftItemClick onLeftItemClick) {
        super();
        addNodeProvider(new HeadProvider());
        addNodeProvider(new CodeDirProvider());
        addNodeProvider(new CodeProvider(onLeftItemClick));
    }


    public interface OnLeftItemClick {
        void onItem(CodeBean codeBean);
    }


    @Override
    protected int getItemType(@NotNull List<? extends BaseNode> data, int position) {
        BaseNode node = data.get(position);
        if (node instanceof HeadNode) {
            return 1;
        } else if (node instanceof CodeDirNode) {
            return 2;
        } else if (node instanceof CodeNode) {
            return 3;
        }
        return -1;
    }

    public static final int EXPAND_COLLAPSE_PAYLOAD = 110;
}

File: app/src/main/java/com/ng/ngleetcode/old/model/code/adapter/NodeTreeAdapter.kt
package com.ng.ngleetcode.old.model.code.adapter

import com.chad.library.adapter.base.BaseNodeAdapter
import com.chad.library.adapter.base.entity.node.BaseNode
import com.ng.ngleetcode.old.model.code.adapter.provider.CodeDirProvider
import com.ng.ngleetcode.old.model.code.adapter.provider.CodeProvider
import com.ng.ngleetcode.old.model.code.adapter.provider.HeadProvider
import com.ng.ngleetcode.old.model.code.bean.CodeDirNode
import com.ng.ngleetcode.old.model.code.bean.CodeNode
import com.ng.ngleetcode.old.model.code.bean.CodeProgressNode

/**
 *    @author : jiangzhengnan.jzn@alibaba-inc.com
 *    @creation   : 2023/05/15
 *    @description   :
 */
class NodeTreeAdapter(onLeftItemClick: OnLeftItemClick) : BaseNodeAdapter() {
    companion object {
        @kotlin.jvm.JvmField
        val EXPAND_COLLAPSE_PAYLOAD = 110
    }

    init {
        addNodeProvider(HeadProvider())
        addNodeProvider(CodeDirProvider())
        addNodeProvider(
            CodeProvider(
                onLeftItemClick
            )
        )
    }

    interface OnLeftItemClick {
        fun onItem(codeBean: CodeNode?)
    }


    override fun getItemType(data: List<BaseNode>, position: Int): Int {
        return when (data[position]) {
            is CodeProgressNode -> {
                3
            }
            is CodeDirNode -> {
                2
            }
            is CodeNode -> {
                1
            }
            else -> 1
        }
    }

}